sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -572,18 +577,20 @@ export function fromRheaMessage(
       props.lockedUntilUtc = new Date(msg.message_annotations[Constants.lockedUntil] as number);
     }
   }
-  if (msg.ttl != null && msg.ttl >= Constants.maxDurationValue - props.enqueuedTimeUtc.getTime()) {
-    props.expiresAtUtc = new Date(Constants.maxDurationValue);
-  } else {
-    props.expiresAtUtc = new Date(props.enqueuedTimeUtc.getTime() + msg.ttl!);
+  if (msg.ttl == null) msg.ttl = Constants.maxDurationValue;
+  if (props.enqueuedTimeUtc) {
```

Comment:
Is props.enqueuedTime ever not set if a message has been received from an actual broker?

Reply:
I hadn't seen such a case, but this gives the flexibility to set the props properly even if the service messes up.